apps/ledger-live-desktop/src/renderer/components/CustomImage/ImageGrayscalePreview.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { Flex } from "@ledgerhq/react-ui";
+import { createCanvas } from "./imageUtils";
+import { ImageBase64Data, ImageDimensions } from "./types";
+import { targetDimensions } from "./shared";
+import ContrastChoice from "./ContrastChoice";
+import { ImageProcessingError } from "@ledgerhq/live-common/customImage/errors";
+
+export type ProcessorPreviewResult = ImageBase64Data & ImageDimensions;
+export type ProcessorRawResult = { hexData: string } & ImageDimensions;
+
+export type ProcessorResult = {
+  previewResult: ProcessorPreviewResult;
+  rawResult: ProcessorRawResult;
+};
+
+const contrasts = [
+  { val: 1, color: "neutral.c70" },
+  { val: 1.5, color: "neutral.c50" },
+  { val: 2, color: "neutral.c40" },
+  { val: 3, color: "neutral.c30" },
+];
+
+function clamp(val: number, min: number, max: number) {
+  return Math.min(max, Math.max(min, val));
+}
+
+function contrastRGB(rgbVal: number, contrastVal: number) {
+  return (rgbVal - 128) * contrastVal + 128;
+}
+
+// simutaneously apply grayscale and contrast to the image
+function applyFilter(
+  imageData: ImageData,
+  contrastAmount: number,
+  dither = true,
+): { imageDataResult: Uint8ClampedArray; hexRawResult: string } {
```

Comment:
feels a bit overkill, was CSS filter considered instead? is it because we want to export the image and save it as grayscale maybe?